pkgs/by-name/et/etherpad-lite/package.nix

```
@@ -0,0 +1,86 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  nix-update-script,
+  pnpm_9,
+  makeWrapper,
+  nodejs,
+}:
+
+let
+  pnpm = pnpm_9;
+in
+stdenv.mkDerivation (finalAttrs: {
+  pname = "etherpad-lite";
+  version = "2.3.0";
+
+  src = fetchFromGitHub {
+    owner = "ether";
+    repo = "etherpad-lite";
+    tag = "v${finalAttrs.version}";
+    hash = "sha256-OomZ1oziEGgLJfYyfkHZOPRHfrrWc4XLAsZA4cn0btA=";
+  };
+
+  patches = [
+    ./outdir.patch
+    # etherpad expects to read and write $out/lib/var/installed_plugins.json
+    # FIXME: this patch disables plugin support
+    ./dont-fail-on-plugins-json.patch
+  ];
+
+  pnpmDeps = pnpm.fetchDeps {
+    inherit (finalAttrs) pname version src;
+    hash = "sha256-nhiPopGLCeCHiqEQ3solwuLwkDnHTH3otbxIJmbuQAA=";
+  };
+
+  nativeBuildInputs = [
+    pnpm.configHook
+    makeWrapper
+  ];
+
+  buildInputs = [
+    nodejs
+  ];
+
+  buildPhase = ''
+    runHook preBuild
+    NODE_ENV="production" pnpm run build:etherpad
+    runHook postBuild
+  '';
+
```

This reduces the closure size from 503M to 110M and makes the build deterministic.

Some explanation: the `.modules.yaml` file contains data that is different for each build (prunedAt date, storeDir) and is not needed at runtime so we can delete it. This file gets regenerated by running `pnpm prune` (which deletes unneeded dev dependencies), so we need to delete at the end, but for some reason if it exists before running `pnpm prune`, less dependencies will be deleted, so run it 2 times.

```suggestion

  preInstall = ''
    # remove unnecessary files
    rm node_modules/.modules.yaml
    pnpm prune --prod --ignore-scripts
    find -type f \( -name "*.d.ts" -o -name "*.map" \) -exec rm -rf {} +

    # remove non-deterministic files
    rm node_modules/.modules.yaml
  '';

```